test/CMakeLists.txt

```diff
@@ -0,0 +1,110 @@
+# Project information.
+cmake_minimum_required ( VERSION 3.13.0 )
+project ( "CoreHTTP unit test"
+          VERSION 1.0.0
+          LANGUAGES C )
+
+# Allow the project to be organized into folders.
+set_property( GLOBAL PROPERTY USE_FOLDERS ON )
+
+# Use C90.
+set( CMAKE_C_STANDARD 90 )
+set( CMAKE_C_STANDARD_REQUIRED ON )
+
+# Do not allow in-source build.
+if( ${PROJECT_SOURCE_DIR} STREQUAL ${PROJECT_BINARY_DIR} )
+    message( FATAL_ERROR "In-source build is not allowed. Please build in a separate directory, such as ${PROJECT_SOURCE_DIR}/build." )
+endif()
+
+# Set global path variables.
+get_filename_component(__MODULE_ROOT_DIR "${CMAKE_CURRENT_LIST_DIR}/.." ABSOLUTE)
+set(MODULE_ROOT_DIR ${__MODULE_ROOT_DIR} CACHE INTERNAL "coreHTTP repository root.")
+
+# Configure options to always show in CMake GUI.
+option( BUILD_CLONE_SUBMODULES
+        "Set this to ON to automatically clone any required Git submodules. When OFF, submodules must be manually cloned."
+        ON )
+
+# Set output directories.
+set( CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin )
+set( CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib )
+set( CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib )
+
+# ====================== Coverity Analysis Configuration =======================
+
+# Include filepaths for source and include.
+include( ${MODULE_ROOT_DIR}/httpFilePaths.cmake )
+
+# Target for Coverity analysis that builds the library.
+add_library( coverity_analysis
+             ${HTTP_SOURCES}
+             ${HTTP_THIRD_PARTY_SOURCES} )
+
+# Build HTTP library target without custom config dependency.
+target_compile_definitions( coverity_analysis PUBLIC HTTP_DO_NOT_USE_CUSTOM_CONFIG=1 )
+
+# HTTP public include path.
+target_include_directories( coverity_analysis PUBLIC ${HTTP_INCLUDE_PUBLIC_DIRS} )
+
+# HTTP private include path.
+target_include_directories( coverity_analysis PRIVATE ${HTTP_INCLUDE_PRIVATE_DIRS} )
```

Comment:
Since we're moving the core_htt_client_private.h file into `source/include`, we may no longer need `HTTP_INCLUDE_PRIVATE_DIRS`. It might be best to just remove it and also update `httpFilePaths.cmake`.

Reply:
It is not moved yet. That is still in PR. I will update depending on which goes in first.